### ISSUE TITLE
DPUScan pdf shows black screen

### ISSUE THREAD

**Issue by sabeesh-mani** [mozilla/pdf.js#10026]
Attach (recommended) or Link to PDF file here:
I am unable to get the official document due to restriction. But I have attached the screenshot. The issue seems to happen for all DPUScan documents of pdf 1.5
![img-20180831-wa0006](https://user-images.githubusercontent.com/42857043/44944859-bb4e9800-ae10-11e8-80b4-d3f5b9f8164e.jpg)

Configuration:
- Web browser and its version: Firefox 45. Also tested on Firefox 52
- Operating system and its version: Windows 7
- PDF.js version: Tested with 2.0.55.0 and 1.10.100
- Is a browser extension: No

Steps to reproduce the problem:
1. Rendering the DPUScan pdf 1.5 version document via pdf.js gives black screen. I also tried opening the same document via online viewer and also ended up with black screen.
2. Attached the screenshot of the problem.

What is the expected behavior? (add screenshot)
PDF should be rendered correctly

What went wrong? (add screenshot)
Black screen is shown

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by timvandermeij** [comment]
From https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md:

```
If the issue is related to errors produced by a specific PDF, please always include the PDF by providing a URL where contributors can download it. Without a PDF for reproduction, such issues will be closed. 
```

Closing since there is nothing we can do without an example PDF file. Perhaps you can make a non-classified PDF file with the same tool to share here, after which we can reopen this.

**Comment by sabeesh-mani** [comment]
I won't be able to share the pdf file here, Can I have your personal emailId? to share the document.

**Comment by THausherr** [comment]
I'd like to have it e-mailed to me at **tilman at snafu dot de**. I'm not with the PDF.js project, I'm a committer on the [Apache PDFBox](https://pdfbox.apache.org/) project (java based) and I look at "problem" files from many projects to improve our quality. Sometimes files that fail here also fail in PDFBox. I can keep your file confidential (I rename such files so it never gets forgotten), or delete it after X days or whatever your wish is.

**Comment by sabeesh-mani** [comment]
Finally managed to get a non-classified pdf file. I have attached the same here. Can someone have a look?
[test.pdf](https://github.com/mozilla/pdf.js/files/2423379/test.pdf)

**Comment by THausherr** [comment]
The file also fails in PDFBox… in java the image has an RGB icc colorspace but in the raster it has only 1 band with 1 bit pixels. One would have to analyze the JPEG2000 image with a good tool that tells what meta data is really there..
[PDFJS-10026-image.zip](https://github.com/mozilla/pdf.js/files/2428635/PDFJS-10026-image.zip)

**Comment by sabeesh-mani** [comment]
Any update we can implement on pdf.js side to render the pdf file?

**Comment by THausherr** [comment]
issue for PDFBox:
https://issues.apache.org/jira/browse/PDFBOX-4326

**Comment by THausherr** [comment]
Here's another exotic PDF file with a JPEG2000 image that can't be rendered with PDF.js; the JPEG2000 image has 4 bits per pixel.
https://issues.apache.org/jira/secure/attachment/12655396/PDFBOX-2204-012411.pdf

**Comment by sabeesh-mani** [comment]
Hello, can you please provide an update?

**Comment by GLuchtenberg** [comment]
Hello! I'm having the same problem :/

**Comment by gschuler** [comment]
Could you please take a look on this? Same problem here!

**Comment by gschuler** [comment]
Actually the whole team depending on this

**Comment by rafaelcaviquioli** [comment]
Please, my head depending on this solution

**Comment by LeonardoPrange** [comment]
Same problem : /

**Comment by THausherr** [comment]
I did some debugging and looked at jpx_stream.js. After jpxImage.parse() the object has 1 tile which is only zeroes. The size is correct, i.e. 2496 x 3512 = 8765952.

(With the other file I attached, the decoded bytes were all 255)

So this suggests that the problem is in the JPEG2000 decoder and not with the colorspace as in PDFBox.

@rafaelcaviquioli this is an open source project and powered by volunteers who may or may not have time. Re "my head depending on this solution": if it is so, then free somebody or several people of your team for a few days and have them debug the JPEG2000 decoder. The show happens in 
https://github.com/mozilla/pdf.js/blob/master/src/core/jpx.js
near "case 0xFF93:".
Compare what is happening with the JPEG2000 specs (see https://jpeg.org/jpeg2000/ ) or compare what is happening to a working JPEG2000 decoder, e.g. the one from Java,
https://github.com/jai-imageio/jai-imageio-jpeg2000/
